pkg/pillar/hardware/smart.go

```diff
@@ -169,20 +225,38 @@ func GetInfoFromNVMeDisk(diskName string) (*types.DiskSmartInfo, error) {
 		return diskInfo, diskInfo.Errors
 	}
 
-	smartTemperature := new(types.DAttrTable)
-	smartTemperature.ID = types.SmartAttrIDTemperatureCelsius
-	smartTemperature.RawValue = int(smartAttr.Temperature)
-	diskInfo.SmartAttrs = append(diskInfo.SmartAttrs, smartTemperature)
+	// processSmartAttributes extracts SMART attributes from the NvmeSMARTLog struct and stores them in diskInfo.
+	// Use reflection to inspect the structure of smartAttr at runtime
+	val := reflect.ValueOf(*smartAttr) // Get the actual values of the fields
+	typ := reflect.TypeOf(*smartAttr)  // Get the metadata (field names, types, etc.)
+
+	// Iterate through all fields in the struct
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)     // Get the value of the current field
+		fieldType := typ.Field(i) // Get metadata about the current field
+
+		// Extract the raw value from the field
+		var rawValue int
+		switch field.Kind() {
+		case reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+			// Directly convert unsigned integer types to int
+			rawValue = int(field.Uint())
+		case reflect.Struct:
+			// This is not accurate but Uint128 is huge so it should be safe.
```

Comment:
Are some attributes strings? Where are structs used?

Reply:
There are no strings NvmeSMARTLog structure, but we do have some slices that we ignore.